File: python/first_test.py
from selenium import webdriver
from selenium.common.exceptions import NoSuchElementException
from selenium.webdriver.common.keys import Keys
import time

# Get browser session
browser = webdriver.Firefox()

# Set implicit wait
browser.implicitly_wait(10)

# Load CW home page
browser.get("http://www.climbingweather.com")

# Assert ClimbingWeather.com in title
assert "ClimbingWeather.com" in browser.title

# Find the search box
search = browser.find_element_by_id("globalSearch")
search.send_keys("Little Cottonwood Canyon" + Keys.RETURN)

# Try to fink link
try:
    area_link = browser.find_element_by_partial_link_text("Little Cottonwood Canyon")
except NoSuchElementException:
    assert 0, "Little Cottonwood Canyon was not in search results"

# Click area link
area_link.click()

# Verify area has 7 forecast days
forecast_days = len(browser.find_elements_by_xpath("//table[contains(@class, 'forecast')]//td[contains(@class, 'day_date')]"))
assert forecast_days == 7, "Expected 7 forecast days but got {0}".format(forecast_days)

# Close browser
browser.close()

File: php/first_test.php
<?php
require_once 'PHPWebDriver/__init__.php';

// This would be the url of the host running the server-standalone.jar
$wd_host = 'http://127.0.0.1:4444/wd/hub'; // this is the default
$web_driver = new PHPWebDriver_WebDriver($wd_host);

// First param to session() is the 'browserName' (default = 'firefox')
// Second param is a JSON object of additional 'desiredCapabilities'

// POST /session
$session = $web_driver->session('firefox');

// Set implicit wait
$session->implicitlyWait(10);

// Open ClimbingWeather.com
$session->open('http://www.climbingweather.com');

// Find search element and submit search
$search = $session->element('id', 'globalSearch');
$search->sendKeys('Little Cottonwood Canyon' . PHPWebDriver_WebDriverKeys::ReturnKey());

// Try to find link
$search = $session->element(PHPWebDriver_WebDriverBy::PARTIAL_LINK_TEXT, 'Little Cottonwood Canyon');
$search->click();

// Check for number of forecast days, should be 7
$days = $session->elements('xpath', "//table[contains(@class, 'forecast')]//td[contains(@class, 'day_date')]");
print count($days);

// Close session
$session->close();
